checker/raw_result.go

```diff
@@ -43,6 +43,7 @@ type RawResults struct {
 	LicenseResults              LicenseData
 	TokenPermissionsResults     TokenPermissionsData
 	CITestResults               CITestData
+	Metadata                    map[string]string
```

Comment:
I just meant sticking it in a struct like all the other fields in `RawResults`. So we can add more to it later in a non-breaking way if we had to. Do you ever envision any metadata that would be non-string?

```
// If you have a better name use it haha.
type MetadataData {
	Metadata map[string]string
}
```

Reply:
Although stepping back, does this metadata really fit with RawResults?

There's already `Metadata []string` in the `ScorecardResult`struct, which is a bit different since it's user supplied.
```
type ScorecardResult struct {
	Repo       RepoInfo
	Date       time.Time
	Scorecard  ScorecardInfo
	Checks     []checker.CheckResult
	RawResults checker.RawResults
	Metadata   []string
}
```

Reply:
Sorry to nitpick, just trying to plan in a way that avoids potential breaking changes in the future to the top level struct

Reply:
no that's fine. I could make it a `Metadata map[string]any`? I would like to keep all values strings in practice, but defining it as a map to `any` would give us flexibility if we want to. Wdut?

Reply:
I still think putting it in a struct gives us the most flexibility to change things later:
```go
// If you have a better name use it haha.
type MetadataData {
	Metadata map[string]string
}
```